crates/cashu/src/nuts/nut02.rs

```diff
@@ -103,26 +136,144 @@ impl Id {
 
     /// [`Id`] from bytes
     pub fn from_bytes(bytes: &[u8]) -> Result<Self, Error> {
-        Ok(Self {
-            version: KeySetVersion::from_byte(&bytes[0])?,
-            id: bytes[1..].try_into()?,
-        })
+        let version = KeySetVersion::from_byte(&bytes[0])?;
+        let id = match version {
+            KeySetVersion::Version00 => IdBytes::V1(bytes[1..].try_into()?),
+            KeySetVersion::Version01 => IdBytes::V2(bytes[1..].try_into()?),
+        };
+        Ok(Self { version, id })
     }
 
+    /// Get the version of the keyset
+    pub fn get_version(&self) -> KeySetVersion {
+        self.version
+    }
+
+    /*
     /// [`Id`] as bytes
     pub fn as_bytes(&self) -> [u8; Self::BYTELEN + 1] {
         let mut bytes = [0u8; Self::BYTELEN + 1];
         bytes[0] = self.version.to_byte();
-        bytes[1..].copy_from_slice(&self.id);
+        match self.id {
```

Comment:
As the whole `as_bytes` function is commented out, maybe we can remove it?

We still have `to_bytes`.